src/Cake.Common/Tools/Cake/CakeRunner.cs

```diff
@@ -150,7 +176,20 @@ protected override string GetToolName()
         /// <returns>The tool executable name.</returns>
         protected override IEnumerable<string> GetToolExecutableNames()
         {
-            return new[] { "Cake.exe" };
+            return new[]
+            {
+#if NETCORE
+                "Cake.dll",
+                "cake",
+                "Cake",
+                "Cake.exe"
+#else
+                "Cake.exe",
+                "cake",
+                "Cake",
+                "Cake.dll"
```

Comment:
Is this required here?

Reply:
Well you could use Cake.CoreCLR  with full if no other found.